docs/proposals/control-plane-config.md

```diff
@@ -17,3 +17,86 @@ option that we will support is log level.
 ## Non-Goals
 
 This proposal is *not* defining a way to dynamically configure the data plane.
+
+## Introduction
+
+The NKG control plane will evolve to have various user-configurable options. These could include, but are not
+limited to, log level, tracing, or metrics. For the best user experience, these options should be able to be
+changed at runtime, to avoid having to restart NKG. The first option that we will allow users to configure is the
+log level. The easiest and most intuitive way to implement a Kubernetes-native API is through a CRD.
+
+## API, Customer Driven Interfaces, and User Experience
+
+The API would be provided in a CRD. An authorized user would interact with this CRD using `kubectl` to `get`
+or `edit` the configuration.
+
+Proposed configuration CRD example:
+
+```yaml
+apiVersion: nginx.gateway.k8s.io/v1beta1
+kind: NGINXControlConfig
+metadata:
+    name: nkg-config
+    namespace: nginx-gateway
+spec:
+    logLevel: info
+    ...
+```
+
+- The CRD would be Namespace-scoped, living in the same Namespace as the controller that it applies to.
+- CRD is initialized and created when NKG is deployed
+- NKG references the name of this CRD via CLI arg, and only watches this CRD
+- If user deletes resource, NKG logs an error and creates an event. Last state is used until CRD is restored.
```

Comment:
would it be better if we use default values instead? This way the control plane will always reconcile to the latest state in the cluster, so it cannot suddenly change behavior because of the restart

Reply:
Sure